packages/dev/docs/pages/react-spectrum/testing.mdx

```diff
@@ -156,3 +158,198 @@ userEvent.type(document.activeElement, 'Pas$w0rd');
+Historically, this most commonly happened when a modals, popovers, or trays is opened or closed as a result of a simulated user action (e.g. opening a Picker's dropdown via click).
+Additionally, we have `requestAnimationFrame` calls in various areas that you will need to also handle by advancing your timers in the tests. This happens most prominently
+in our virtualized components due to the internals of our Virtualizer. In Jest, this can be handled by calling `act(() => jest.runAllTimers());` but you may require more precise control
+depending on the other time sensitive behavior you are testing. Please see [Jest's timer docs](https://jestjs.io/docs/timer-mocks) or the equivalent docs of your test frameworks for more information on how to do so.
+It is also a good idea to run all timers to completion after each test case to avoid any left over transitions or timeouts that a component may have setup during its lifecycle.
+
+```tsx
+afterEach(() => {
+  act(() => jest.runAllTimers());
+});
+```
+
+Consider adding a `act(() => jest.runAllTimers());` after your simulated user interaction if you run into a test failure that looks like the following:
+
+```
+TestingLibraryElementError: Unable to find an accessible element with the role "listbox"
+```
+
+If you are using real timers instead, you can await for popovers to appear or for a particular state of your app to be reached. If you are using React Testing Library, you can perform a `waitFor` query
+to wait for a dialog to appear:
+
+```tsx
+await waitFor(() => {
+  expect(getByRole('dialog')).toBeInTheDocument();
+});
+```
+
+### Desktop vs Mobile
+
+Components like ComboBox and Picker render a different experience on mobile versus desktop. If you need to test mobile vs desktop for you app or want to make sure the React Spectrum
+components are rendered only against a specific experience, you'll need to mock the window screen width accordingly. We currently use a
+[screen width of 700px](https://github.com/adobe/react-spectrum/blob/5e487532e3ca4714513fb6ab64f7a78d9d1ca281/packages/%40react-spectrum/utils/src/useIsMobileDevice.ts#L15-L23) as the breakpoint
+between mobile and desktop, so you should mock the `width` getter of `window.screen` to a value greater or less than this value accordingly. You may use <TypeLink links={testUtilDocs.links} type={testUtilDocs.exports.simulateMobile} /> and
+<TypeLink links={testUtilDocs.links} type={testUtilDocs.exports.simulateDesktop} /> from the `@react-spectrum/test-utils` package to handle this width mocking for you in Jest:
+
+The example below shows how you would set this up in Jest.
+
+```tsx
+import {simulateDesktop, simulateMobile} from '@react-spectrum/test-utils';
+
+beforeAll(() => {
+  simulateMobile();
+});
+
+afterAll(() => {
+  jest.restoreAllMocks();
+});
+```
+
+TODO: this section assumes that we will turn off virtualization for in test environments, update if aren't going ship with that
```

Comment:
Just a reminder to handle this before merging

Reply:
Yup, will add a comment below so I don't forget